ttnn/cpp/ttnn/operations/matmul/device/matmul_op.hpp

```diff
@@ -142,15 +137,17 @@ using MatmulProgramConfig = std::variant<
 
 
 struct Matmul {
-    const std::optional<const MatmulProgramConfig> program_config;
-    bool bcast_batch;
-    const MemoryConfig output_mem_config;
-    const DataType output_dtype;
-    const DeviceComputeKernelConfig compute_kernel_config;
-    const bool untilize_out;
-    const std::optional<const CoreCoord> user_core_coord;
-    const std::optional<const UnaryWithParam> user_fused_activation;
-    const std::optional<const bool> user_run_batched;
+    const std::optional<const MatmulProgramConfig> program_config = std::nullopt;
+    const std::optional<bool> bcast_batch = std::nullopt;
+    const MemoryConfig output_mem_config = operation::DEFAULT_OUTPUT_MEMORY_CONFIG;
+    const std::optional<DataType> output_dtype = std::nullopt;
+    const std::optional<DeviceComputeKernelConfig> compute_kernel_config = std::nullopt;
+    const bool untilize_out = false;
+    const std::optional<const CoreCoord> user_core_coord = std::nullopt;
+    const std::optional<const UnaryWithParam> user_fused_activation = std::nullopt;
+    const bool user_run_batched = false;
```

Comment:
what is user_run_batched used for?

Reply:
It's used to maintain functional parity from before when
- a program config is not provided
- a user specified core grid is not provided and
- input tensor a is sharded

When those three criteria are met, the parameter is used to determine the type of program config, whether MatmulMultiCoreReuseProgramConfig or MatmulMultiCoreReuseMultiCastProgramConfig/MatmulMultiCoreReuseMultiCast1DProgramConfig

ttnn.linear() always has it set to false.
ttnn.matmul() has it set to true if input b is batched.